### PR TITLE
Update gdal tools

### PR DESCRIPTION
**Pull request by Alanamosse**
- Update gdal's version from 2.4.0 to 3.0.0
  - update @ VERSION@
  - changes in output test files

- Update ogr2ogr and ogrinfo to use the shp composite datatype
  - Update commands blocs in xml
  - add new tests
  - add new testfiles

### CODE REVIEW

**Comment by Alanamosse** [comment]
Hi @bgruening !
New update for Gdal =) It should enable the use of shp datatype.
However Travis uses the Galaxy release 19.01 to set up the tests, we should upgrade to 19.05 in [travis.ylm](https://github.com/galaxyecology/tools-ecology/blob/master/.travis.yml) so Galaxy recognise shp files.

I will open a PR for this.

**Comment by Alanamosse** [comment]
The corresponding [PR](https://github.com/galaxyecology/tools-ecology/pull/13)

File: tools/gdal/.shed.yml [hunk 1]
```diff
@@ -81,6 +81,13 @@ repositories:
       - gdal_macros.xml
       - test-data/custom.geo.json
       - test-data/ogrinfo_test1_out2.txt
+      - test-data/shp_compositefiles/shapefile.html
```
Comment (bgruening) [inline tools/gdal/.shed.yml]: btw. @Alanamosse have you ever tried to exclude files? 

https://github.com/kpbioteam/ewas_galaxy/blob/master/tools/minfi/.shed.yml#L14

Would that work? You could test this with the Test ToolShed.

File: tools/gdal/ogr2ogr.xml [hunk 2]
```diff
@@ -2,63 +2,106 @@
...
-        <param type="select" name="f" label="Conversion format" value=""
-            help="Select the format to convert your input file. Currently supported : Shapefile, GeoJSON, GML, KML and WFS. The shapefiles outputs shp, shx, dbf and prj will populate a datacollection.">
-            <option value="ESRI Shapefile">ESRI Shapefile</option> 
+        <param type="data" name="input" format="shp,xml,json,txt" label="Gdal supported input file" help="http://www.gdal.org/ogr_formats.html"/>
+        <param type="select" name="f" label="Conversion format" value="" help="Select the format to convert your input file. Currently supported : Shapefile, GeoJSON, GML, KML and WFS.">
+            <option value="ESRI Shapefile">ESRI Shapefile</option>
             <option value="GeoJSON">GEOJSON</option> 
             <option value="GML">GML</option>
-            <option value="KML">KML</option>
+	    <option value="KML">KML</option>
```
Comment (bgruening) [inline tools/gdal/ogr2ogr.xml]: indentation?

File: tools/gdal/ogr2ogr.xml [hunk 3]
```diff
@@ -2,63 +2,106 @@
...
-                <data name="dbf" format="binary" from_work_dir="*.dbf"/>
-                <data name="prj" format="txt" from_work_dir="*.prj"/>
-       </collection>
+        </data>
     </outputs>
-    
     <tests>
-        <test> <!-- ogr2ogr -f KML custom.geo.kml custom.geo.json -->
-            <param name="input" value="custom.geo.json"/>
+        <test>
```
Comment (bgruening) [inline tools/gdal/ogr2ogr.xml]: You could use that trick: https://github.com/galaxyproject/tools-iuc/pull/2444/files#diff-182bc3778630f0a87cb6db71e73edf6aR380

And test your number of outputs and with this your filters.

**Comment by bgruening** [comment]
I think you need to merge master into this branch now or rebase against master.

File: tools/gdal/ogr2ogr.xml (hunk 2 above)
Comment (Alanamosse) [reply tools/gdal/ogr2ogr.xml]: Indentation ;)

**Comment by bgruening** [comment]
No clue why this is now failing ... but it seems the test data is not matching anymore :(

**Comment by Alanamosse** [comment]
Ok after a research it seems that the `.dbf` file saves the "Date of last update; formatted as YYMMDD" 
So this test is valid one day, I change the `compare` method with `sim_size` and a small delta.

File: tools/gdal/.shed.yml (hunk 1 above)
Comment (bgruening) [reply tools/gdal/.shed.yml]: ping @Alanamosse ... but can wait for later.

Comment (Alanamosse) [reply tools/gdal/.shed.yml]: Yes I know the feature but not sure to understand here.
The `auto_repo` takes the needed testfiles and the `exclude` prevents the others to be in the repo as well ?
 I give it a try in test toolshed

Comment (bgruening) [reply tools/gdal/.shed.yml]: The idea was just, maybe its easier to exclude the few files that you don't want to have in ... Not sure. Just a thought.

**Pull request merged**